File: DAA/Min Max/min_Max.cpp
#include<iostream>
using namespace std;

// int min;
// int max;

void minMax(int left, int right, int arr[], int &max, int &min){
    int min1, max1;
    if(left == right){
        max = arr[left];
    }
    else if(left == right -1){
        if(arr[left] < arr[right]){
            min = arr[left];
            max = arr[right];
        }
        else{
            min = arr[right];
            max = arr[left];  
        }
    }
    else{
        int mid = (left+right)/2;
        minMax(left,mid,arr,max,min);
        max1 = max, min1 = min;
        minMax(mid+1,right,arr,max,min);
        if(max1 > max){
            max = max1;
        }
        if(min1 < min){
            min=min1;
        }
    }
}

int main()
{
    cout<<"Enter array length: ";
    int lenArr; cin>>lenArr;
    int *arr = new int[lenArr];
    int min,max;
    for(int i=1; i<=lenArr; i++){
        cout<<"Enter number: "; cin>>arr[i];
    }
    minMax(1,lenArr,arr,max,min);
    cout<<"Maximum: "<<max<<" Minimum: "<<min<<endl;

}

File: DAA/Min Max/minMax.java

public class minMax {
    static int min, max;

    static void rMinMax(int arr[],int i,int j){
        
        int max1;
        int min1;

        if(i == j){
            min = arr[i];
            max = arr[j];
        }
        else if (i == j-1){
            if(arr[i] > arr[j-1]){
                max = arr[i];
                min = arr[j];
            }
            else{
                max = arr[j];
                min = arr[i];
            }
        }
        else{
            int mid = (i + j)/2;
            rMinMax(arr, i, mid);
            max1=max; min1=min;
            rMinMax(arr, mid+1, j);
            if(max1 > max1){
                max=max1;
            }
            if(min1<min){
                min=min1;
            }
            
        }
    }

    public static void main(String[] args){
        int[] arr = {22, 13, -5, -8, 15, 60, 17, 31, 47};
        rMinMax(arr,0, arr.length-1);

        System.out.println("Maximum value: "+max+" Minimum value: "+min);
    }

}
